framework/src/base/EigenProblem.C

```diff
@@ -162,3 +162,12 @@ EigenProblem::converged()
 {
   return _nl_eigen->converged();
 }
+
+bool
+EigenProblem::isNonlinearEigenvalueSolver()
+{
+  return solverParams()._eigen_solve_type == Moose::EST_NONLINEAR_POWER ||
+         solverParams()._eigen_solve_type == Moose::EST_MF_NONLINEAR_POWER ||
+         solverParams()._eigen_solve_type == Moose::EST_MONOLITH_NEWTON ||
+         solverParams()._eigen_solve_type == Moose::EST_MF_MONOLITH_NEWTON;
```

Comment:
`solverParams()._eigen_solve_type` is set at the beginning of` solve()`, and so I will get rid of `_is_nonlinear_eigenvalue_solver`

Thanks, @andrsd